lms/djangoapps/certificates/models.py

```diff
@@ -558,6 +563,9 @@ def certificate_status(generated_certificate):
         if generated_certificate.status == CertificateStatuses.downloadable:
             cert_status['download_url'] = generated_certificate.download_url
 
+        if generated_certificate.course_id.certificate_available_date < datetime.datetime.utcnow():
```

Comment:
After looking at this, I think I might want to change the status on the generated certificate too, but I can't find any other cases where the generated certificate status is set so I'm not sure where would be a good place to set it.